### DISCUSSION TITLE
Indexing Dynamic Vectors?

### DISCUSSION THREAD

**Discussion by jpmccu** [milvus-io/milvus#27933] [Q&A and General discussion]
Hi all,

Is it possible to index a dynamic vector?

For instance, if we add a field dynamically by name, and we always insert vectors into it, can we build an index over it? What are the rules there? How does it handle vectors of different lengths?

**Comment by filip-halt** [comment]
Vectors of different lengths cannot be used in the same collection. While you can upsert new vectors of the same length, you will want them to be from the same model throughout as the vectors need to be within the same latent space.

**Comment by jpmccu** [comment]
What if they're the same length by convention, if not schema?

**Reply by yhmo** [reply]
I suppose you are talking about the "[dynamic field](https://milvus.io/docs/dynamic_schema.md)".

The dynamic field stores data in JSON format. JSON list can store vectors. So, you can store vectors with different lengths in the dynamic field.
But milvus doesn't support index for the dynamic field, and milvus cannot perform ANN search on JSON list. You can use ["json_contains" or "in"](https://milvus.io/docs/json_data_type.md#JavaScript-Object-Notation-JSON) keywords to test element existence of a JSON list as a filtering condition, but you are not able to do ANN search to dynamic field like the vector field.
